File: CR1111/d.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
#include <utility>

using namespace std;


void solve() {
    int n, q;
    cin >> n >> q;

    vector<pair<long long, int>> elements(n);

    for (int i = 0; i < n; ++i) {
        long long value;
        cin >> value;
        elements[i] = {value, i};
    }

    sort(elements.begin(), elements.end());

    int mask = 0;

    for (int target = 0; target < n; ++target) {
        int original = elements[target].second;
        mask |= original ^ target;
    }

    if (mask == 0) {
        cout << 0 << '\n';
        return;
    }

    int answer = 1 << (31 - __builtin_clz(mask));

    cout << answer << '\n';
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    while (t--) {
        solve();
    }

    return 0;
}

File: CR1111/d.py
def solve():
    n, q = map(int, input().split())
    arr = list(map(int, input().split()))
    order = sorted(range(n), key=lambda x: arr[x])
    mask = 0
    
    for target, original in enumerate(order):
        mask |= original ^ target
        
    if mask == 0:
        print(0)
    else:
        print(1 << (mask.bit_length() - 1))

if __name__ == "__main__":
     t = int(input())
     for _ in range(t):
         solve()